parlai/agents/seq2seq/modules.py

```diff
@@ -309,278 +340,199 @@ def forward(self, xs):
             encoder_output, _ = pad_packed_sequence(encoder_output,
                                                     batch_first=True)
         if self.dirs > 1:
-            # take elementwise max between forward and backward hidden states
-            # NOTE: currently using max, but maybe should use Linear
+            # project to decoder dimension by taking sum of forward and back
             if isinstance(self.rnn, nn.LSTM):
-                hidden = (hidden[0].view(-1, self.dirs, bsz, self.hsz).max(1)[0],
-                          hidden[1].view(-1, self.dirs, bsz, self.hsz).max(1)[0])
+                hidden = (hidden[0].view(-1, self.dirs, bsz, self.hsz).sum(1),
+                          hidden[1].view(-1, self.dirs, bsz, self.hsz).sum(1))
             else:
-                hidden = hidden.view(-1, self.dirs, bsz, self.hsz).max(1)[0]
+                hidden = hidden.view(-1, self.dirs, bsz, self.hsz).sum(1)
 
-        return encoder_output, hidden
+        return hidden, encoder_output, attn_mask
```

Comment:
intentional swap?

Reply:
nit TBH I think encoder_output, hidden, attn_mask is better.

Reply:
swap was intentional because later
```
hidden = encoder_states[0]
attn_params = (encoder_states[1], encoder_states[2])
```
but I could change that to [1] and ([0], [2])

it does match the rnn output more closely